kernel/src/table_changes/log_replay/tests.rs

```diff
@@ -124,31 +130,34 @@ async fn configuration_fails() {
 #[tokio::test]
 async fn incompatible_schema() {
     let engine = Arc::new(SyncEngine::new());
-    let mut mock_table = MockTable::new();
+    let mut mock_table = LocalMockTable::new();
 
     // The original schema has two fields: `id` and value.
     let schema = get_schema().project(&["id"]).unwrap();
     let schema_string = serde_json::to_string(&schema).unwrap();
     mock_table
-        .commit(&[Metadata {
-            schema_string,
-            configuration: HashMap::from([
-                ("delta.enableChangeDataFeed".to_string(), "true".to_string()),
-                (
-                    "delta.enableDeletionVectors".to_string(),
-                    "true".to_string(),
-                ),
-            ]),
-            ..Default::default()
-        }
-        .into()])
+        .commit(
+            [Metadata {
+                schema_string,
+                configuration: HashMap::from([
+                    ("delta.enableChangeDataFeed".to_string(), "true".to_string()),
+                    (
+                        "delta.enableDeletionVectors".to_string(),
+                        "true".to_string(),
+                    ),
+                ]),
+                ..Default::default()
+            }
+            .into()]
+            .into_iter(),
```

Comment:
Changing `commit` to take `Iterator` instead of `IntoIterator` caused a lot of call site bloat...